framework/DataObjects/XDataSet.py

```diff
@@ -918,11 +918,11 @@ def _fromCSVXML(self,fName):
     # check if we have anything from which to read
     try:
       meta,_ = xmlUtils.loadToTree(fName+'.xml')
+      self.raiseADebug('Reading metadata from "{}.xml"'.format(fName))
+      haveMeta = True      
     except IOError:
       haveMeta = False
-    finally:
-      self.raiseADebug('Reading metadata from "{}.xml"'.format(fName))
-      haveMeta = True
+      
```

Comment:
I don't think this change actually changes anything ...

Reply:
It does...

```finally:``` is always executed and you were setting the ```haveMeta = True``` in there. 
So even if the ```except``` was setting the ```haveMeta = False``` the finally was always resetting it.

Reply:
Oh, it should definitely be "else" instead of "finally".  True.  We should avoid putting too many things in the "try" though, as that is prone to misguided error statements.